docs/tutorials/sparse/csr.md

```diff
@@ -0,0 +1,258 @@
+operator `[]` to copy to an existing array.
+
+```python
+a = mx.nd.ones((2,2)).tostype('csr')
+b = a.copy()
+c = mx.nd.sparse.zeros('csr', (2,2))
+c[:] = a
+d = mx.nd.sparse.zeros('csr', (2,2))
+a.copyto(d)
+{'b is a': b is a, 'b.asnumpy()':b.asnumpy(), 'c.asnumpy()':c.asnumpy(), 'd.asnumpy()':d.asnumpy()}
+```
+
+* If the storage types of source array and destination array doesn't match,
+the storage type of destination array won't change when copying with `copyto` or
+the slice operator `[]`.
+
+```python
+e = mx.nd.sparse.zeros('csr', (2,2))
+f = mx.nd.sparse.zeros('csr', (2,2))
+g = mx.nd.ones(e.shape)
+e[:] = g
+g.copyto(f)
+{'e.stype':e.stype, 'f.stype':f.stype}
+```
+
+## Indexing and Slicing
+
+We can slice a CSRNDArray on axis 0 with operator `[]`, which copies the slices and returns a new CSRNDArray.
+
+```python
+a = mx.nd.array(np.arange(6).reshape(3,2)).tostype('csr')
+b = a[1:2]
+c = a[:].asnumpy()
+{'b':b, 'c':c}
+```
+
+## Sparse Operators and Storage Type Inference
+
+* Operators that have specialized implementation for sparse arrays can be accessed in ``mx.nd.sparse``.
+You can read the [mxnet.ndarray.sparse API documentation](mxnet.io/api/python/ndarray.html) to find
+what sparse operators are available.
+
+```python
+shape = (3, 4)
+data = [7, 8, 9]
+indptr = [0, 2, 2, 3]
+indices = [0, 2, 1]
+a = mx.nd.sparse.csr_matrix(data, indptr, indices, shape) # a csr matrix as lhs
+rhs = mx.nd.ones((4, 1))      # a dense vector as rhs
+out = mx.nd.sparse.dot(a, rhs)  # invoke sparse dot operator specialized for dot(csr, dense)
+{'out':out}
+```
+
+* For any sparse operator, the storage type of output array is inferred based on inputs. You can either read
+the documentation or inspect the `stype` attribute of output array to know what storage type is inferred:
+
+```python
+b = a * 2  # b will be a CSRNDArray since zero multiplied by 2 is still zero
+c = a + 1  # c will be a dense NDArray
+{'b.stype':b.stype, 'c.stype':c.stype}
+```
+
+* For operators that don't specialize in sparse arrays, we can still use them with sparse inputs with some performance penalty.
+What happens is that MXNet will generate temporary dense inputs from sparse inputs so that the dense operators can be used.
+Warning messages will be printed when such storage fallback event happens.
+
+```python
+d = mx.nd.log(a) # warnings will be printed
+{'a.stype':a.stype, 'd':d} # stype of a is not changed
+```
+
+## Loading LibSVM Data
+
+Sparse data stored in libsvm file format can be loaded with [mx.io.LibSVMIter](https://mxnet.incubator.apache.org/versions/master/api/python/io.html#mxnet.io.LibSVMIter).
+Note that the indices are expected to be zero-based instead of one-based.
+
+## Advanced Topics
+
+### GPU Support
+
+By default, CSRNDArray operators are executed on CPU. In MXNet, GPU support for CSRNDArray is experimental
+with few sparse operators such as cast_storage and dot.
```

Comment:
Add: with only a few sparse operators such as...

Reply:
I  used `few` instead of `a few` because we only have 2 operators supported for GPU.. I can change it if `only a few ` is more accurate.